### PR TITLE
Whitesource integration 

### PR DESCRIPTION
**Pull request by sfc-gh-japatel**
- I have tested integration in Travis. 
- This includes changes in deploy.sh so that I can try our jenkins build can also run whitesource script. Hence I have changed `mvn deploy` to `mvn package` temporarily

Here is an example build: https://travis-ci.org/github/snowflakedb/snowflake-ingest-java/builds/700199431

### CODE REVIEW

File: scripts/run_whitesource.sh [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+#
+# Run whitesource for components which need versioning
+
+# SCAN_DIRECTORIES is a comma-separated list (as a string) of file paths which contain all source code and build artifacts for this project
+echo "PWD:"${PWD}
+SCAN_DIRECTORIES=${PWD}
+echo "SCAN_DIRECTORIES:"$SCAN_DIRECTORIES
+
+
+if [[ "$PWD" == *travis* ]]; then
```
Comment (sfc-gh-stakeda) [inline scripts/run_whitesource.sh:11]: https://docs.travis-ci.com/user/environment-variables/#default-environment-variables
```
if [[ -n "$TRAVIS"]]; then
```
This is more reliable.

File: scripts/run_whitesource.sh [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+# SCAN_CONFIG="${SCAN_DIRECTORIES}/wss-generated-file.config"
+
+# SCAN_CONFIG is the path to your whitesource configuration file
+SCAN_CONFIG="scripts/wss-java-maven-agent.config"
+echo "SCAN_CONFIG:"$SCAN_CONFIG
+
+echo "[INFO] Running wss.sh for ${PRODUCT_NAME}-${PROJECT_NAME} under ${SCAN_DIRECTORIES}"
+
+if [ "$GIT_BRANCH" != "$PROD_BRANCH" ]; then
+    echo "First Branch"
```
Comment (sfc-gh-stakeda) [inline scripts/run_whitesource.sh:62]: echo "PR"

File: scripts/run_whitesource.sh [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        -d ${SCAN_DIRECTORIES} \
+        -projectVersion ${PROJECT_VERSION}
+    ERR=$?
+    if [[ "$ERR" != "254" && "$ERR" != "0" ]]; then
+        echo "failed to run whitesource scanning with feature branch"
+        # if you want to fail the build when failing to run whitesource with projectName feature branch, please use exit 1 
+        # exit 1
+    fi 
+else
+    echo "SECOND BRANCH"
```
Comment (sfc-gh-stakeda) [inline scripts/run_whitesource.sh:76]: echo "Production"

File: scripts/run_whitesource.sh (hunk 1 above)
Comment (sfc-gh-japatel) [reply scripts/run_whitesource.sh:11]: Will do!

File: scripts/run_whitesource.sh (hunk 2 above)
Comment (sfc-gh-japatel) [reply scripts/run_whitesource.sh:62]: Sorry my bad, I didnt clean this up!

File: scripts/run_whitesource.sh (hunk 1 above)
Comment (sfc-gh-stakeda) [reply scripts/run_whitesource.sh:11]: See https://github.com/snowflakedb/snowflake-connector-python/blob/master/ci/wss.sh#L45-L54 fo Travis checks.

I don't think TRAVIS_PULL_REQUEST becomes an empty if not PR.

Comment (sfc-gh-japatel) [reply scripts/run_whitesource.sh:11]: Yes, I saw my build using false as a value. Will change it, thank you!
Sorry I couldnt have verified this before generating a PR. (Unless a draft)

**Pull request merged**